### PR TITLE
Fix for Desktop build break

### PR DESCRIPTION
**Pull request by briansull**

### CODE REVIEW

**Comment by briansull** [comment]
PTAL @CarolEidt @sandreenko

File: src/jit/compiler.h [hunk 1]
```diff
@@ -1082,8 +1082,9 @@ class LclVarDsc
 #ifdef FEATURE_HFA
         assert(lvIsHfa());
         return HfaTypeFromElemKind(_lvHfaElemKind);
-#endif // FEATURE_HFA
+#else
```
Comment (sandreenko) [inline src/jit/compiler.h]:
```
# else // !FEATURE_HFA
#endif // !FEATURE_HFA
```

File: src/jit/compiler.h [hunk 2]
```diff
@@ -1642,8 +1643,9 @@ struct fgArgTabEntry
     {
 #ifdef FEATURE_HFA
         return HfaTypeFromElemKind(_hfaElemKind);
-#endif // FEATURE_HFA
+#else
```
Comment (sandreenko) [inline src/jit/compiler.h]: the same here.

File: src/jit/compiler.h [hunk 3]
```diff
@@ -1773,10 +1775,12 @@ struct fgArgTabEntry
                 // Round up in case of odd HFA count.
                 size = (size + 1) >> 1;
             }
+#ifdef FEATURE_SIMD
             else if (hfaType == TYP_SIMD16)
             {
                 size <<= 1;
             }
+#endif
```
Comment (sandreenko) [inline src/jit/compiler.h]: `#endif // FEATURE_SIMD`.

File: src/jit/compiler.h [hunk 4]
```diff
@@ -1642,8 +1643,9 @@ struct fgArgTabEntry
     {
 #ifdef FEATURE_HFA
         return HfaTypeFromElemKind(_hfaElemKind);
-#endif // FEATURE_HFA
+#else
         return TYP_UNDEF;
```
Comment (sandreenko) [inline src/jit/compiler.h]: Did we have an unreachable return and it was not threated as error by CoreCLR compiler?

Comment (mikedn) [reply src/jit/compiler.h]: Yes, I hit the same thing in my /W4 PR.

File: src/jit/compiler.h (hunk 1 above)
Comment (CarolEidt) [reply src/jit/compiler.h]: In general, when the `#ifdef`s are small non-nested like this, we don't need to comment them:
```
Do specify the macro name in a comment at the end of the closing #endif of a long or nested #if/#ifdef.
```
https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#75-commenting-ifdefs
and personally I find that it makes it quite cluttered.

That said, if there *is* a closing comment, there should similarly be a comment on the `#else` (i.e. `!FEATURE_HFA` in this case), and the `#endif` should have the "!"

Comment (briansull) [reply src/jit/compiler.h]: I also don't think that for one line ifdefs we need topcomment the #else

File: src/jit/compiler.h (hunk 2 above)
Comment (briansull) [reply src/jit/compiler.h]: Same as above

**Comment by briansull** [comment]
Merge - Comment only change

**Pull request merged**